source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c

```diff
@@ -453,6 +473,29 @@ BaseType_t xGetPhyLinkStatus( void )
     return xReturn;
 }
 /*-----------------------------------------------------------*/
+NetworkInterface_t * pxFillInterfaceDescriptor( BaseType_t xEMACIndex,
+                                                NetworkInterface_t * pxInterface )
+{
+    static char pcName[ 17 ];
+
+/* This function pxFillInterfaceDescriptor() adds a network-interface.
+ * Make sure that the object pointed to by 'pxInterface'
+ * is declared static or global, and that it will remain to exist. */
+
+    snprintf( pcName, sizeof( pcName ), "eth%u", ( unsigned ) xEMACIndex );
+
+    memset( pxInterface, '\0', sizeof( *pxInterface ) );
+    pxInterface->pcName = pcName;                    /* Just for logging, debugging. */
+    pxInterface->pvArgument = ( void * ) xEMACIndex; /* Has only meaning for the driver functions. */
```

Comment:
Originally, I thought of storing an object here, a pointer to a struct. But most drivers, like this one, store a short number of the EMAC-in-use.